src/h5web/visualizations/index.test.ts

```diff
@@ -1,559 +1,408 @@
-const baseDataset = {
-  id: 'dataset',
-  collection: HDF5Collection.Datasets as const,
-};
-const baseDatatype = {
-  id: 'datatype',
-  collection: HDF5Collection.Datatypes as const,
-};
-
-const nxDataGroup = {
-  ...baseGroup,
-  attributes: [
-    { name: 'NX_class', value: 'NXdata' },
-    { name: 'signal', value: 'my_signal' },
-  ] as HDF5Attribute[],
-  links: [
-    {
-      class: HDF5LinkClass.Hard as const,
-      title: 'my_signal',
-      collection: HDF5Collection.Datasets,
-      id: baseDataset.id,
-    },
-  ],
-};
-
-function supportsEntity(
-  vis: Vis,
-  entity: HDF5Entity,
-  metadata = {} as HDF5Metadata
-): boolean {
-  return VIS_DEFS[vis].supportsEntity(entity, metadata);
+import {
+  makeSimpleDataset,
+  intType,
+  makeGroup,
+  makeDatatype,
+  advancedType,
+  makeDataset,
+  scalarShape,
+  makeSimpleShape,
+  stringType,
+  makeStrAttr,
+  makeMetadata,
+  withAttributes,
+  makeDatasetHardLink,
+  makeNxDataGroup,
+} from '../providers/mock/data';
+
+const datasetIntScalar = makeDataset('dataset_int', intType, scalarShape);
+const datasetFltScalar = makeDataset('dataset_flt', intType, scalarShape);
+const datasetInt0D = makeSimpleDataset('dataset_int_0d', intType, []);
+const datasetInt1D = makeSimpleDataset('dataset_int_1d', intType, [5]);
+const datasetInt2D = makeSimpleDataset('dataset_int_2d', intType, [5, 3]);
+const datasetStr1D = makeSimpleDataset('dataset_str_1d', stringType, [5]);
+const datasetStr2D = makeSimpleDataset('dataset_str_2d', stringType, [5, 3]);
+const datasetFlt3D = makeSimpleDataset('dataset_flt_3d', intType, [5, 3, 1]);
+
+const groupEmpty = makeGroup('group_empty');
+const datatypeAdvanced = makeDatatype('datatype_advanced', advancedType);
+
+function makeSupportFn(
+  vis: Vis
+): (entity: HDF5Entity, metadata?: HDF5Metadata) => boolean {
+  return (entity: HDF5Entity, metadata = {} as HDF5Metadata) =>
+    VIS_DEFS[vis].supportsEntity(entity, metadata);
 }
 
 describe('Visualization definitions', () => {
   describe('Vis.Raw', () => {
-    it('should support any dataset', () => {
-      const supportedDatasets: HDF5Entity[] = [
-        { ...baseDataset, type: '', shape: { class: HDF5ShapeClass.Null } },
-      ];
+    const supportsEntity = makeSupportFn(Vis.Raw);
 
-      supportedDatasets.forEach((dataset) => {
-        expect(supportsEntity(Vis.Raw, dataset)).toBe(true);
-      });
+    it('should support any dataset', () => {
+      expect(supportsEntity(datasetInt1D)).toBe(true);
     });
 
     it('should not support group and datatype', () => {
-      const unsupportedEntities: HDF5Entity[] = [
-        { ...baseGroup },
-        { ...baseDatatype, type: '' },
-      ];
-
-      unsupportedEntities.forEach((entity) => {
-        expect(supportsEntity(Vis.Raw, entity)).toBe(false);
-      });
+      expect(supportsEntity(groupEmpty)).toBe(false);
+      expect(supportsEntity(datatypeAdvanced)).toBe(false);
```

Comment:
The new test syntax is a lot clear:

- no type casts (i.e. `as ...`) -- the generated mock objects are all valid HDF5 objects;
- no loop, instead each expect is written explicitly.